stl/src/wlocale.cpp

```diff
@@ -10,37 +10,33 @@ _STD_BEGIN
 #pragma warning(disable : 4074)
 #pragma init_seg(compiler)
 
-
-using _Wtraits  = char_traits<wchar_t>;
-using _Winiter  = istreambuf_iterator<wchar_t, _Wtraits>;
-using _Woutiter = ostreambuf_iterator<wchar_t, _Wtraits>;
-
 // facets associated with C categories
-#define ADDFAC(Facet, cat, ptrimp, ptrloc)                                            \
-    if ((_CATMASK(Facet::_Getcat()) & cat) == 0) {                                    \
-        ;                                                                             \
-    } else if (ptrloc == nullptr) {                                                   \
-        ptrimp->_Addfac(new Facet(lobj), Facet::id);                                  \
-    } else {                                                                          \
-        ptrimp->_Addfac((locale::facet*) &_STD use_facet<Facet>(*ptrloc), Facet::id); \
+#define ADDFAC(Facet, cat, ptrimp, ptrloc)                                                                  \
+    if ((_CATMASK(Facet::_Getcat()) & cat) == 0) {                                                          \
+        ;                                                                                                   \
+    } else if (ptrloc == nullptr) {                                                                         \
+        ptrimp->_Addfac(new Facet(lobj), Facet::id);                                                        \
+    } else {                                                                                                \
+        ptrimp->_Addfac(                                                                                    \
+            const_cast<locale::facet*>(static_cast<const locale::facet*>(&_STD use_facet<Facet>(*ptrloc))), \
+            Facet::id);                                                                                     \
     }
 
 // moved from locale to ease subsetting
 using _Tw1  = ctype<wchar_t>;

```

Comment:
More yuck (no change requested)

Reply:
The fact that the macro is repeated across 3 source files is triple yuck. This code is fragile (damaging the locale IDs was the worst I ever broke the STL in production), but I'd like to clean this up further in the future.

I don't even really understand what the "subsetting" is talking about.